src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/AbstractCloudFoundryDeployer.java

```diff
@@ -147,9 +151,21 @@ int diskQuota(AppDeploymentRequest request) {
 		return (int) ByteSizeUtils.parseToMebibytes(withUnit);
 	}
 
-	String buildpack(AppDeploymentRequest request) {
-		return Optional.ofNullable(request.getDeploymentProperties().get(BUILDPACK_PROPERTY_KEY))
-			.orElse(this.deploymentProperties.getBuildpack());
+	Set<String> buildpacks(AppDeploymentRequest request) {
+		// TODO: When 'buildpack' setting gets removed due to deprecation,
+		//       change this logic not ot fallback to it if 'buildpacks'
+		//       is used.
+		String buidpacksValue = request.getDeploymentProperties().get(BUILDPACKS_PROPERTY_KEY);
+		String buidpackValue = request.getDeploymentProperties().get(BUILDPACK_PROPERTY_KEY);
+		if (buidpacksValue != null) {
+			return StringUtils.commaDelimitedListToSet(buidpacksValue);
+		} else if (buidpackValue != null) {
```

Comment:
Looks like you need to change your IDE formatter settings - I see the else on the same line